threadpoolctl.py

```diff
@@ -105,27 +105,25 @@ class LibController(ABC):
     """
 
     @final
-    def __init__(self, *, filepath=None, prefix=None):
+    def __init__(self, *, filepath=None, prefix=None, parent=None):
         """This is not meant to be overriden by subclasses."""
+        self.parent = parent
         self.prefix = prefix
         self.filepath = filepath
         self.dynlib = ctypes.CDLL(filepath, mode=_RTLD_NOLOAD)
         self.version = self.get_version()
         self.set_additional_attributes()
 
-    @final
     def info(self):
-        """Return relevant info wrapped in a dict
-
-        This is not meant to be overriden by subclasses.
-        """
```

Comment:
we now need to be able to override this since the flexiblas controller has dynamic properties (so we can't rely on `vars`)